runtime/compiler/optimizer/J9EstimateCodeSize.cpp

```diff
@@ -88,7 +88,7 @@ class NeedsPeekingHeuristic
             int32_t len;
             const char *sig = p->getTypeSignature(len);
             if (i >= argInfo->getNumArgs() ||            //not enough slots in argInfo
-               *sig != 'L' ||                            //primitive arg
+               *sig != 'L' || *sig != 'Q' ||             //primitive arg
```

Comment:
I think this line should read as follows:

```
(*sig != 'L' && *sig != 'Q') ||
```

Otherwise, it will always be true.